src/bin/cargo-msrv.rs

```diff
@@ -68,13 +73,13 @@ fn init_and_run(ctx: &Context) -> Result<ExitCode, InstanceError> {
 
     tracing::info!("storyteller channel created");
 
-    let output_format = ctx.output_format();
+    let output_format = opts.shared_opts.user_output_opts.output_format;
```

Comment:
I hope this is fine and there is no way `output_format()` could differ from `opts.shared_opts.user_output_opts.output_format`.

Reply:
We should be able to find a way to use the context.

In cargo-msrv, the context can be considered a resolved, merged and immutable set of inputs (and are split per subcommand; it's an enum). The (...)Opts are CLI arguments. 

In the future I will also add (more) environment variables and potentially support for a config file. These input sets will be merged according to some precedence. So, concretely, if at some point we have:

- `CargoMsrvOpts`
- `CargoMsrvEnvVars`
- `CargoMsrvConfigFile`

Then after resolving and merging these, the remaining inputs for the program is the `Context`.

(Thanks for the PR 😄, I'll do a full review later)